libraries/outputs.rb

```diff
@@ -24,6 +24,35 @@ def self.configure_outputs(node) # rubocop:disable Metrics/PerceivedComplexity,
         else
           [CernerSplunk.my_cluster(node)]
         end.each do |(cluster, bag)|
+          if bag['indexer_discovery'] == true
+            Chef::Log.warn "Configured ['receivers'] in cluster #{cluster} will be ignored since ['indexer_discovery'] is set to true." if bag['receivers']
+
+            indexer_discovery_settings = ((bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['outputs_configs']) || {}).reject do |k, _|
+              k.start_with?('_cerner_splunk')
+            end
+            output_stanzas["indexer_discovery:#{cluster}"] = indexer_discovery_settings
+
+            if bag['master_uri'].nil? || bag['master_uri'].empty?
+              Chef::Log.warn "master_uri is missing in the cluster databag: #{cluster}"
+            else
+              output_stanzas["indexer_discovery:#{cluster}"]['master_uri'] = bag['master_uri']
+            end
+            encrypt_password = CernerSplunk::ConfTemplate::Transform.splunk_encrypt node: node
+
+            pass =
+              if bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['pass4SymmKey']
+                bag['indexer_discovery_settings']['pass4SymmKey']
+              else
+                'changeme'
+              end
+
+            output_stanzas["indexer_discovery:#{cluster}"]['pass4SymmKey'] = CernerSplunk::ConfTemplate.compose encrypt_password, CernerSplunk::ConfTemplate::Value.constant(value: pass)
+            output_stanzas["tcpout:#{cluster}"] = {}
+            output_stanzas["tcpout:#{cluster}"]['indexerDiscovery'] = cluster
+            output_stanzas["tcpout:#{cluster}"]['useACK'] = true
```

Comment:
I suspect this should be configurable... the tradeoff is a possibility of duplicated data versus a possibility of lost data from a forwarder.

Reply:
I hardcoded it because I remember that I couldn't get the indexer discovery to work during vagrant testing without this attribute set.

Reply:
Tested this again today morning and it worked without setting useACK. It didn't work before because the pass4symmkey I configured was different on the cluster master and the forwarder (hence I removed the configured pass4symmkey from the databag). Added the option to configure tcpout attributes a07479b681b8d30108c0393832ff5eea872339e8